src/gurt/telemetry.c

```diff
@@ -1776,31 +1943,45 @@ int d_tm_add_metric(struct d_tm_node_t **node, int metric_type, char *sh_desc,
 		temp->dtn_metric->dtm_stats->dtm_min = UINT64_MAX;
 	}
 
-	buff_len = strnlen(sh_desc, D_TM_MAX_SHORT_LEN);
+	buff_len = 0;
+	if (sh_desc != NULL)
+		buff_len = strnlen(sh_desc, D_TM_MAX_SHORT_LEN);
 	if (buff_len == D_TM_MAX_SHORT_LEN) {
 		rc = -DER_EXCEEDS_PATH_LEN;
 		goto failure;
 	}
-	buff_len += 1; /** make room for the trailing null */
-	temp->dtn_metric->dtm_sh_desc = d_tm_shmalloc(buff_len);
-	if (temp->dtn_metric->dtm_sh_desc == NULL) {
-		rc = -DER_NO_SHMEM;
-		goto failure;
+
+	if (buff_len > 0) {
+		buff_len += 1; /** make room for the trailing null */
+		temp->dtn_metric->dtm_sh_desc = d_tm_shmalloc(buff_len);
+		if (temp->dtn_metric->dtm_sh_desc == NULL) {
+			rc = -DER_NO_SHMEM;
+			goto failure;
+		}
+		strncpy(temp->dtn_metric->dtm_sh_desc, sh_desc, buff_len);
+	} else {
+		temp->dtn_metric->dtm_sh_desc = NULL;
 	}
-	strncpy(temp->dtn_metric->dtm_sh_desc, sh_desc, buff_len);
 
-	buff_len = strnlen(lng_desc, D_TM_MAX_LONG_LEN);
+	buff_len = 0;
+	if (lng_desc != NULL)
+		buff_len = strnlen(lng_desc, D_TM_MAX_LONG_LEN);
 	if (buff_len == D_TM_MAX_LONG_LEN) {
 		rc = -DER_EXCEEDS_PATH_LEN;
 		goto failure;
 	}
-	buff_len += 1; /** make room for the trailing null */
-	temp->dtn_metric->dtm_lng_desc = d_tm_shmalloc(buff_len);
-	if (temp->dtn_metric->dtm_lng_desc == NULL) {
-		rc = -DER_NO_SHMEM;
-		goto failure;
+
+	if (buff_len > 0) {
```

Comment:
looks like you could probably make a helper function to reduce duplication of code between here and above regarding handling of shmalloc call

Reply:
I left this alone for now, only because I wasn't sure that my helper function implementation was actually useful beyond what I had.  I know it wasn't a blocking issue for you, so if it is still important, we can talk about it more.